homeworks/week2/Quyen-Ngo.java

```diff
@@ -0,0 +1,39 @@
+import java.util.HashSet;
+import java.util.Scanner;
+
+public class Quyen-Ngo {
+
+	public static void main(String[] args) {
+
+		//Time complexity : O(n)
+		//Space complexity: O(n)		
+
+
+		Scanner scan = new Scanner(System.in);
+		String s = scan.nextLine();
+		HashSet<Character> hs = new HashSet<Character>();
+		char[] arr = s.toCharArray();
+		
+		int count = 0;
+		for(int i = 0; i < arr.length; i++) {
+			if (!(hs.add(arr[i]))) {
+				arr[i]= ' ';
+				count++;
+			}
+		}
+		
+		char[] array = new char[arr.length-count];
```

Comment:
An easier-to-understand approach would be using a `List` or a `StringBuffer` to add unique characters one by one. Another downside of this solution is that `' '` is possible to appear as a character in the input string

Reply:
so many helpful comments, thank you so much.